pkg/agent/flowexporter/connections/conntrack_ovs.go

```diff
@@ -0,0 +1,174 @@
+var _ ConnTrackDumper = new(connTrackOvsCtl)
+
+type connTrackOvsCtl struct {
+	nodeConfig   *config.NodeConfig
+	serviceCIDR  *net.IPNet
+	ovsctlClient ovsctl.OVSCtlClient
+}
+
+func NewConnTrackOvsAppCtl(nodeConfig *config.NodeConfig, serviceCIDR *net.IPNet, ovsctlClient ovsctl.OVSCtlClient) *connTrackOvsCtl {
+	if ovsctlClient == nil {
+		return nil
+	}
+	return &connTrackOvsCtl{
+		nodeConfig,
+		serviceCIDR,
+		ovsctlClient,
+	}
+}
+
+// DumpFlows uses "ovs-appctl dpctl/dump-conntrack" to dump conntrack flows in the Antrea ZoneID.
+func (ct *connTrackOvsCtl) DumpFlows(zoneFilter uint16) ([]*flowexporter.Connection, error) {
+	conns, err := ct.ovsAppctlDumpConnections(zoneFilter)
+	if err != nil {
+		klog.Errorf("Error when dumping flows from conntrack: %v", err)
+		return nil, err
+	}
+
+	filteredConns := filterAntreaConns(conns, ct.nodeConfig, ct.serviceCIDR, zoneFilter)
+	klog.V(2).Infof("Flow exporter considered flows: %d", len(filteredConns))
+
+	return filteredConns, nil
+}
+
+func (ct *connTrackOvsCtl) ovsAppctlDumpConnections(zoneFilter uint16) ([]*flowexporter.Connection, error) {
+	// Dump conntrack using ovs-appctl dpctl/dump-conntrack
+	cmdOutput, execErr := ct.ovsctlClient.RunAppctlCmd("dpctl/dump-conntrack", false, "-m", "-s")
+	if execErr != nil {
+		return nil, fmt.Errorf("error when executing dump-conntrack command: %v", execErr)
+	}
+
+	// Parse the output to get the flows
+	antreaConns := make([]*flowexporter.Connection, 0)
+	outputFlow := strings.Split(string(cmdOutput), "\n")
+	var err error
+	for _, flow := range outputFlow {
+		conn := flowexporter.Connection{}
+		flowSlice := strings.Split(flow, ",")
+		isReply := false
+		inZone := false
+		for _, fs := range flowSlice {
+			// Indicator to populate reply or reverse fields
+			if strings.Contains(fs, "reply") {
+				isReply = true
+			}
+			if !strings.Contains(fs, "=") {
+				// Proto identifier
+				conn.TupleOrig.Protocol, err = ip.LookupProtocolMap(fs)
+				if err != nil {
+					klog.Errorf("Unknown protocol to convert to ID: %s", fs)
+					continue
+				}
+				conn.TupleReply.Protocol = conn.TupleOrig.Protocol
+			} else if strings.Contains(fs, "src") {
+				fields := strings.Split(fs, "=")
+				if !isReply {
+					conn.TupleOrig.SourceAddress = net.ParseIP(fields[len(fields)-1])
+				} else {
+					conn.TupleReply.SourceAddress = net.ParseIP(fields[len(fields)-1])
+				}
+			} else if strings.Contains(fs, "dst") {
+				fields := strings.Split(fs, "=")
+				if !isReply {
+					conn.TupleOrig.DestinationAddress = net.ParseIP(fields[len(fields)-1])
+				} else {
+					conn.TupleReply.DestinationAddress = net.ParseIP(fields[len(fields)-1])
+				}
+			} else if strings.Contains(fs, "sport") {
+				fields := strings.Split(fs, "=")
+				val, err := strconv.Atoi(fields[len(fields)-1])
+				if err != nil {
+					klog.Errorf("Conversion of sport: %s to int failed", fields[len(fields)-1])
```

Comment:
for this log and the next ones, remove the ":", it doesn't really make sense

Reply:
done